jms/src/main/scala/akka/stream/alpakka/jms/javadsl/package.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.jms
+
+import akka.NotUsed
+import akka.annotation.InternalApi
+import akka.stream
+import akka.stream.alpakka.jms
+import akka.stream.alpakka.jms.scaladsl.JmsConnectorState._
+import akka.stream.javadsl.Source
+
+package object javadsl {
+
+  @InternalApi private[javadsl] def transformConnectorState(
+      source: stream.scaladsl.Source[jms.scaladsl.JmsConnectorState, NotUsed]
+  ): Source[JmsConnectorState, NotUsed] =
+    source.map {
+      case Disconnected => JmsConnectorState.Disconnected
```

Comment:
I guess the Scala states could contain the Java state and offer `asJava`.

Reply:
Ok, that introduce a dependency from the scaladsl to the javadsl, would that be fine for you?

Reply:
Yes, even if the dependency goes the other direction for most cases.

Reply:
Implemented with https://github.com/akka/alpakka/pull/1252/commits/6a406b743c6209de04bd21e14985a1ea98e512e5